test/extended/cmd.sh

```diff
@@ -54,88 +54,89 @@ oadm policy add-role-to-group view system:authenticated -n default
 
 install_registry
 wait_for_registry
-DOCKER_REGISTRY=`oc get service/docker-registry -n default -o jsonpath='{.spec.clusterIP}:{.spec.ports[0].port}'`
+docker_registry="$( oc get service/docker-registry -n default -o jsonpath='{.spec.clusterIP}:{.spec.ports[0].port}' )"
 
 
 echo "[INFO] Running newapp extended tests"
-oc login ${MASTER_ADDR} -u new-app -p password --certificate-authority=${MASTER_CONFIG_DIR}/ca.crt
+oc login "${MASTER_ADDR}" -u new-app -p password --certificate-authority="${MASTER_CONFIG_DIR}/ca.crt"
 oc new-project new-app
 oc delete all --all
 
 # create a local-only docker image for testing
 # image is removed in cleanup()
 tmp=$(mktemp -d)
-pushd $tmp
+pushd "${tmp}"
 cat <<-EOF >> Dockerfile
 	FROM scratch
 	EXPOSE 80
 EOF
 docker build -t test/scratchimage .
 popd
-rm -rf $tmp
+rm -rf "${tmp}"
 
 # ensure a local-only image gets a docker image(not imagestream) reference created.
-[ "$(oc new-app test/scratchimage~https://github.com/openshift/ruby-hello-world.git --strategy=docker -o yaml |& tr '\n' ' ' | grep -E "from:\s+kind:\s+DockerImage\s+name:\s+test/scratchimage:latest\s+")" ]
+os::cmd::expect_success 'oc new-project test-scratchimage'
+os::cmd::expect_success 'oc new-app test/scratchimage~https://github.com/openshift/ruby-hello-world.git --strategy=docker'
```

Comment:
This is a significant change.  You just started running a build for this.  The only reason is to be able to run a template?  pipe the dry-run through `oc convert` with a template.